### PR TITLE
UX improvements for unauthorized users

### PR DESCRIPTION
**Pull request by legrego**
As discussed, this does not account for future use cases (e.g. granular app privs)

### CODE REVIEW

**Comment by legrego** [comment]
@kobelb for your viewing pleasure

File: x-pack/plugins/security/public/views/management/edit_role/components/privileges/kibana/space_aware_privilege_form.test.tsx [hunk 1]
```diff
@@ -37,6 +37,7 @@ const buildProps = (customProps: any = {}) => {
         name: 'Marketing',
       },
     ],
+    userProfile: { hasCapability: () => true },
```
Comment (kobelb) [inline x-pack/plugins/security/public/views/management/edit_role/components/privileges/kibana/space_aware_privilege_form.test.tsx:40]: Can we add a test where this is set to false to ensure we're rendering the callout below?

File: x-pack/plugins/spaces/server/lib/spaces_client.ts [hunk 2]
```diff
@@ -26,6 +26,19 @@ export class SpacesClient {
     this.request = request;
   }
 
+  public async canEnumerateSpaces(): Promise<boolean> {
```
Comment (kobelb) [inline x-pack/plugins/spaces/server/lib/spaces_client.ts:33]: Can we add tests for this?

**Review by kobelb — APPROVED** [review]
LGTM!

**Comment by legrego** [comment]
woohoo! Feel free to merge at your convenience

**Comment by legrego** [comment]
Closing in favor of https://github.com/elastic/kibana/pull/22982, which is a cherry-pick of these commits

**Pull request closed without merging**